docs/contributing/documentation_guidelines.md

```diff
@@ -211,35 +247,177 @@ Rendered output:
 -   Auto generated label (preferred) [](my_awesome_section)
 -   [label for section](my_awesome_section)
 
-You can read more about other methods of linking to, and creating references in the MyST parser docs section on [Targets and cross-referencing](https://myst-parser.readthedocs.io/en/stable/syntax/cross-referencing.html).
+You can read more about other methods of linking to, and creating references in the MyST-Parser docs section on [](inv:myst#syntax/cross-referencing).
 
 #### Intersphinx links (external docs)
 
-Due to the large amount of documentation links to Django's Sphinx documentation, we have added the integration with this via intersphinx references.
+Due to the large amount of links to external documentation (especially Django), we have added the integration via intersphinx references. This is configured via [](inv:sphinx:std:confval#intersphinx_mapping) in the `docs/conf.py` file. This allows you to link to specific sections of a project's documentation and catch warnings when the target is no longer available.
+
+Markdown example:
 
 ```md
-You can select widgets from [Django's form widgets](inv:django#ref/forms/widgets)
+You can select widgets from [Django's form widgets](inv:django#ref/forms/widgets).
 ```
 
+reStructuredText example:
+
 ```rst
-    This parameter allows you to specify a :doc:`Django form widget <django:ref/forms/widgets>` to use instead of the default widget for this field type.
+You can select widgets from :doc:`Django form widget <django:ref/forms/widgets>`.
+```
+
+The format for a Sphinx link in Markdown is `inv:key:domain:type#name`. The `key`, `domain`, and `type` are optional, but are recommended to avoid ambiguity when there are multiple targets with the same `name`.
+
+If the `name` contains a space, you need to wrap the whole link in angle brackets `<>`.
+
+```md
+See Django's docs on [](<inv:django:std:label#topics/cache:template fragment caching>).
```

Comment:
I didn't know that this was possible, so TIL

https://github.com/wagtail/wagtail/pull/9603#discussion_r1246402586